components/size_aware_image.jsx

```diff
@@ -170,6 +182,79 @@ export default class SizeAwareImage extends React.PureComponent {
             />
         );
 
+        // copyLink, download are two buttons overlayed on image preview
+        // copyLinkTooltip, downloadTooltip are tooltips for the buttons respectively.
+        // if linkCopiedRecently is true, defaultMessage would be 'Copy Link', else 'Copied!'
+
+        const copyLinkTooltip = (
+            <Tooltip
+                id='copy-link-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='single_image_view.copy_link_tooltip'
+                    defaultMessage={this.state.linkCopiedRecently ? 'Copied' : 'Copy link'}
+                />
+            </Tooltip>
+        );
+        const copyLink = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={copyLinkTooltip}
+                rootClose={true}
+            >
+                <button
+                    key='copy-link'
+                    className={classNames('style--none', 'size-aware-image__copy_link', {
+                        'size-aware-image__copy_link--recently_copied': this.state.linkCopiedRecently,
+                    })}
+                    aria-label='Copy Link to Asset'
```

Comment:
This should be translatable like the other text in this component. Since this is a class component, our best bet for this is `Utils.localizeMessage(<id>, <defaultMessage>)`